hack/jenkins/pipelines/bootkube-e2e/Jenkinsfile

```diff
@@ -0,0 +1,81 @@
+pipeline {
+  agent {
+    kubernetes {
+      cloud 'kubernetes'
+      label "${JOB_NAME}-${BUILD_NUMBER}"
+      containerTemplate {
+        name 'default'
+        image 'colemickens/bootkube-e2e:1520055055'
+        ttyEnabled true
+        command 'cat'
+      }
+    }
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    ansiColor('xterm')
+    timestamps()
+    skipDefaultCheckout(true)
+  }
+  environment {
+    CLUSTER_NAME="${JOB_NAME}-${BUILD_NUMBER}"
+    GOPATH = "${WORKSPACE}"
+    WORKDIR = "${WORKSPACE}/src/github.com/kubernetes-incubator/bootkube"
+    KUBECONFIG = "${WORKSPACE}/src/github.com/kubernetes-incubator/bootkube/hack/quickstart/cluster/auth/kubeconfig"
+    IDENT = "${WORKSPACE}/src/github.com/kubernetes-incubator/bootkube/hack/quickstart/cluster/auth/id_rsa"
+
+    AWS_CRED = credentials('aws')
+  }
+  stages {
+    stage('checkout') {
+      steps {
+        container("default") {
+          sh """mkdir -p "${WORKDIR}";"""
+          dir("${WORKDIR}") {
+            checkout scm
+          }
+        }
+      }
+    }
+    stage('build') {
+      steps {
+        container("default") {
+          dir("${WORKDIR}") {
+            sh "make all"
+          }
+        }
+      }
+    }
+    stage('deploy') {
+      steps {
+        container("default") {
+          dir("${WORKDIR}") {
+            script {
+              env["ACCESS_KEY_ID"] = "${AWS_CRED_USR}"
+              env["ACCESS_KEY_SECRET"] = "${AWS_CRED_PSW}"
+            }
+            sh "./hack/jenkins/scripts/tqs-up.sh"
+          }
+        }
+      }
+    }
+    stage('e2e') {
+      steps {
+        container("default") {
+          dir("${WORKDIR}") {
+            sh "./hack/jenkins/scripts/e2e.sh"
```

Comment:
we should execute these scripts with bash, since the shebangs are bash.

Reply:
nevermind... sh is part of the dsl

Reply:
The default shell for Jenkins is `/bin/bash` and `sh` is actually a pipeline command, rather than the unix `sh`. AFAICT everything is working as expected, including bash expansions.